lib/query-interface.js

```diff
@@ -497,10 +497,13 @@ QueryInterface.prototype.removeIndex = function(tableName, indexNameOrAttributes
 };
 
 QueryInterface.prototype.insert = function(instance, tableName, values, options) {
+  options = options || {};
+  options.hasTrigger = !!(instance && instance.__options && instance.__options.hasTrigger);
```

Comment:
Aah, true - in that case `instance && instance.Model.hasTrigger` should do the trick here